api-service/src/controllers/DataIngestion/DataIngestionController.ts

```diff
@@ -0,0 +1,77 @@
+import { Request, Response } from "express";
+import * as _ from 'lodash';
+import dataIngestorSchema from "./DataIngestion.json";
+import { schemaValidation } from "../../services";
+import { ResponseHandler } from "../../helpers/ResponseHandler";
+import { send } from "../../connections/kafkaConnection";
+import { Dataset } from "../../models/Dataset";
+
+const dataIn = async (req: Request, res: Response) => {
+    try {
+        const datasetId = getDatasetId(req);
+        schemaValidation(req.body, dataIngestorSchema)
+        req.body = { ...req.body.data, dataset: datasetId };
+        const datasetRecord = await getDatasetRecord(datasetId)
+        const validData = await validation(req.body, datasetId);
+        await send(validData, _.get(datasetRecord, "dataValues.dataset_config.entry_topic", ""))
+        ResponseHandler.successResponse(req, res, { status: 200, data: { message: "Data ingestion successful." } });
+    }
+    catch (err: any) {
+        ResponseHandler.errorResponse(err, req, res);
+    }
+}
+
+const getDatasetId = (req: Request) => {
+    const datasetId = req.params.datasetId.trim()
+    if (!_.isEmpty(datasetId)) {
+        return datasetId
+    }
+    else {
+        throw {
+            "message": "datasetId parameter in url cannot be empty",
+            "statusCode": 400,
+            "errCode": "BAD_REQUEST"
+        };
+    }
+}
+
+export const getDatasetRecord = async (datasetId: string) => {
+    const dataset = await Dataset.findOne({
+        where: {
+            id: datasetId,
+        },
+    });
+    if (dataset !== null) {
+        return dataset
+    }
+    else {
+        throw {
+            "message": "Dataset with id not found",
+            "statusCode": 404,
+            "errCode": "BAD_REQUEST"
+        }
+    }
+}
+
+const validation = async (data: any, datasetId: string) => {
```

Comment:
Simplify this logic to two-step 
1. For single event
2. With extraction config

Reply:
solved